src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -189,29 +189,56 @@ private function replaceByShortType(Tokens $tokens, TypeAnalysis $type, array $u
      */
     private function getTypes(Tokens $tokens, int $index, int $endIndex): iterable
     {
-        $index = $typeStartIndex = $typeEndIndex = $tokens->getNextMeaningfulToken($index - 1);
-        $type = $tokens[$index]->getContent();
-
+        $skipNextYield = false;
+        $typeStartIndex = $typeEndIndex = null;
+        $type = null;
         while (true) {
-            $index = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$index]->isGivenKind(CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_OPEN)) {
+                $index = $tokens->getNextMeaningfulToken($index);
+                $typeStartIndex = $typeEndIndex = null;
+                $type = null;
+
+                continue;
+            }
 
-            if ($tokens[$index]->isGivenKind([CT::T_TYPE_ALTERNATION, CT::T_TYPE_INTERSECTION])) {
-                yield $type => [$typeStartIndex, $typeEndIndex];
+            if (
+                $tokens[$index]->isGivenKind([CT::T_TYPE_ALTERNATION, CT::T_TYPE_INTERSECTION, CT::T_DISJUNCTIVE_NORMAL_FORM_TYPE_PARENTHESIS_CLOSE])
+                || $index > $endIndex
+            ) {
+                if (!$skipNextYield && null !== $typeStartIndex) {
+                    $origCount = \count($tokens);
+
+                    yield $type => [$typeStartIndex, $typeEndIndex];
+
+                    $endIndex += \count($tokens) - $origCount;
```

Comment:
Hmmmm, do we rely on `$tokens` being modified outside of this method by iterator's consumer? It's pretty hard to grasp, because at first sight `\count($tokens) - $origCount` is 0 since `$origCount` referes to the same `count()`...  Why it's like that?

Reply:
I tried first another way, but this seems fine - live update was used also before this PR, but the code was working by coincidence... I spent quite some time on this and now it should be working in all cases safely.

Reply:
```
$origCount = \count($tokens);

yield $type => [$typeStartIndex, $typeEndIndex];

$endIndex += \count($tokens) - $origCount;
```

should be self explanatory the count can be change... the comment you propose is present more or less one line below

Reply:
Yeah, I saw that comment but it's less helpful IMHO (not only because of how it's worded, but because of placing). I'll leave decision to you 😉.